src/middlewares/accommodation/imageUploader.js

```diff
@@ -0,0 +1,112 @@
+import { v2 as cloudinary } from 'cloudinary'
+import 'dotenv/config';
+
+cloudinary.config({
+    cloud_name: process.env.CLOUDINARY_CLOUD_NAME,
+    api_key: process.env.CLOUDINARY_API_KEY,
+    api_secret: process.env.CLOUDINARY_API_SECRET
+});
+
+const imageUploader = {}
+
+imageUploader.accommodationCreation = (req, res, next) => {
+    if((Object.keys(req.inputFiles).includes("accommodation_image"))){
+        try{
+            cloudinary.uploader.upload(
+                req.inputFiles.accommodation_image.filepath, 
+                { folder: "winners-c8-bn-be/accommodations/facilities/" },
+                (error, result) => {
+    
+                    req.inputFiles.cloudinaryOutput = result;
+                    req.body.images_links = [result.url];
+                    next()
+                }
+            );
+        }
+        catch(err){
+            return res.status(500).json({
+                error: `Failed to upload the image, due to this error: ${err}`,
+              });
+        }
+    }
+    else{ 
+        return res.status(400).json({
+            error: "the accommodation facility's image file is not provided.",
+          });
+    }
+}
+imageUploader.accommodationUpdate = (req, res, next) => {
+    if((Object.keys(req.inputFiles).includes("accommodation_image"))){
+        try{
+            cloudinary.uploader.upload(
+                req.inputFiles.accommodation_image.filepath, 
+                { folder: "winners-c8-bn-be/accommodations/facilities/" },
```

Comment:
we don't need to store the files in our source codes

Reply:
UPDATE: I chose to change that line to `{ folder: `winners-c8-bn-be/${process.env.NODE_ENV}/accommodations/facilities/` },` so that that `process.env.NODE_ENV` variable will help to separate files uploaded due to tests, development or production.